code/modules/mob/living/living_say.dm

```diff
@@ -277,17 +277,17 @@ GLOBAL_LIST_INIT(department_radio_keys, list(
 			var/empty_indexes = get_empty_held_indexes() //How many hands the player has empty
 			if(length(empty_indexes) == 1 || !mute.get_bodypart(BODY_ZONE_L_ARM) || !mute.get_bodypart(BODY_ZONE_R_ARM))
 				message = stars(message)
-			if(length(empty_indexes) == 0)//Both hands full, can't sign
-				to_chat(src, "<span class='warning'>You can't sign with your hands full!</span.?>")
+			if(length(empty_indexes) == 0 || length(empty_indexes) == 1 && !mute.get_bodypart(BODY_ZONE_L_ARM) || length(empty_indexes) == 1 && !mute.get_bodypart(BODY_ZONE_R_ARM))//All existing hands full, can't sign
+				mute.visible_message("<span class='warning'>[src] tries to sign, but can't with [src.p_their()] hands full!</span.?>")
 				return FALSE
 			if(!mute.get_bodypart(BODY_ZONE_L_ARM) && !mute.get_bodypart(BODY_ZONE_R_ARM))//Can't sign with no arms!
 				to_chat(src, "<span class='warning'>You can't sign with no hands!</span.?>")
 				return FALSE
 			if(mute.handcuffed)//Can't sign when your hands are cuffed, but can at least make a visual effort to
 				mute.visible_message("<span class='warning'>[src] tries to sign, but can't with [src.p_their()] hands cuffed!</span.?>")
 				return FALSE
-			if(mute.has_status_effect(STATUS_EFFECT_PARALYZED))
-				to_chat(src, "<span class='warning'>You can't sign in this state!</span.?>")
+			if(mute.cant_sign)//Upgraded to our own variable that can be added anywhere!
+				to_chat(src, "<span class='warning'>You can't sign at the moment!</span.?>")
```

Comment:
I feel like this could be better handled with a combination of paralyzed and knockdown, what was the full reason this was needed? I don't wanna bloat stuff up too much.

Reply:
The correct way here would be too add a `HANDS_BLOCKED` trait, dismembering it off the incapacitated and stun mobility traits. I had this planned.
This variable suffers from being way too specific, there's behavior the can be generalized here. And also from being a negative statement, giving you a double negation if you flip it with `!`

Reply:
I can see why this would be a big issue, yea. I'll revert the variable changes, seeing as you already had things planned. I swapped it for multiple reasons, including swapping flashes stopping you from signing to making stun batons stop you from signing, as well as so I could possibly add an ability to silence sign users using Mime's Bane. Having a `HANDS_BLOCKED` trait would be _very_ helpful with all of this.

Reply:
When you said you had this planned, I assumed you meant in the future, and as such I did not do anything. Should I attempt to add it in myself, or would it be best to wait for you to do it?

Reply:
It's added already.

Reply:
I swapped it over when you made the review with it, no worries.